lightning/src/ln/peers/handshake/mod.rs

```diff
@@ -392,3 +392,374 @@ impl PeerHandshake {
+	}
+
+	// Outbound::Uninitiated -> AwaitingActTwo (extra bytes in argument)
+	#[test]
+	fn peer_handshake_outbound_uninitiated_to_awaiting_act_two_nonempty_input() {
+		let mut test_ctx = TestCtx::new();
+
+		// TODO: process_act() should error if state does not use vec, but it is non-empty
+		assert_matches!(test_ctx.outbound_handshake.process_act(&[1]).unwrap(), (Some(Act::One(_)), None));
+		assert_matches!(test_ctx.outbound_handshake.state, Some(HandshakeState::AwaitingActTwo(_)));
+		assert_eq!(test_ctx.outbound_handshake.get_remote_pubkey(), Some(test_ctx.inbound_public_key));
+	}
+
+	// Inbound::AwaitingActOne -> Error (input too small)
+	#[test]
+	fn peer_handshake_new_inbound_awaiting_act_one_input_too_small() {
+		let mut test_ctx = TestCtx::new();
+
+		assert_eq!(test_ctx.inbound_handshake.process_act(&[]).err(), Some(String::from("need at least 50 bytes")));
+	}
+
+	// Inbound::AwaitingActOne -> AwaitingActThree (excess bytes)
+	// TODO: This should error early if we receive act3 data prior to sending act2
+	#[test]
+	fn peer_handshake_new_inbound_awaiting_act_one_input_too_large() {
+		let mut test_ctx = TestCtx::new();
+		let mut act1 = do_process_act_or_panic!(test_ctx.outbound_handshake, &[]);
+		act1.extend_from_slice(&[1]);
+
+		assert_matches!(test_ctx.inbound_handshake.process_act(&act1).unwrap(), (Some(Act::Two(_)), None));
+		assert_matches!(test_ctx.inbound_handshake.state, Some(HandshakeState::AwaitingActThree(_)));
+		assert!(test_ctx.inbound_handshake.get_remote_pubkey().is_none());
+	}
+
+	// Inbound::AwaitingActOne -> Error (bad version byte)
+	#[test]
+	fn peer_handshake_new_inbound_awaiting_act_one_bad_version() {
+		let mut test_ctx = TestCtx::new();
+		let mut act1 = do_process_act_or_panic!(test_ctx.outbound_handshake, &[]);
+		// Set bad version byte
+		act1[0] = 1;
+
+		assert_eq!(test_ctx.inbound_handshake.process_act(&act1).err(), Some(String::from("unexpected version")));
+	}
+
+	// Inbound::AwaitingActOne -> Error (invalid hmac)
+	#[test]
+	fn peer_handshake_new_inbound_awaiting_act_invalid_hmac() {
+		let mut test_ctx = TestCtx::new();
+		let mut act1 = do_process_act_or_panic!(test_ctx.outbound_handshake, &[]);
+		// corrupt the ciphertext
+		act1[34] = 0;
+
+		assert_eq!(test_ctx.inbound_handshake.process_act(&act1).err(), Some(String::from("invalid hmac")));
+	}
+
+	// Inbound::AwaitingActOne -> Error (invalid remote ephemeral key)
+	#[test]
+	fn peer_handshake_new_inbound_awaiting_act_invalid_remote_ephemeral_key() {
+		let mut test_ctx = TestCtx::new();
+		let mut act1 = do_process_act_or_panic!(test_ctx.outbound_handshake, &[]);
+		// corrupt the ephemeral public key
+		act1[1] = 0;
+
+		assert_eq!(test_ctx.inbound_handshake.process_act(&act1).err(), Some(String::from("invalid remote ephemeral public key")));
+	}
+
+	// Inbound::AwaitingActOne -> AwaitingActThree
+	#[test]
+	fn peer_handshake_new_inbound_awaiting_act_one_to_awaiting_act_three() {
```

Comment:
For consistency with the other state transition tests, move the successful test case before the failure cases. Or vice versa, if your prefer.